extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperiment.js

```diff
@@ -0,0 +1,144 @@
+define('VisualEditorTourExperiment', ['jquery', 'wikia.loader', 'wikia.mustache', 'wikia.tracker'],
+	function ($, loader, mustache, tracker) {
+		'use strict';
+
+		var track = tracker.buildTrackingFunction({
+				category: 've-editing-tour',
+				trackingMethod: 'analytics'
+			});
+
+		function Tour(tourConfig, labelPrefix) {
+			this.labelPrefix = labelPrefix;
+			this.tourConfig = tourConfig;
+			this.steps = [];
+		}
+
+		Tour.prototype.start = function() {
+			this.step = -1;
+			loader({
+				type: loader.MULTI,
+				resources: {
+					mustache: 'extensions/wikia/VisualEditorTourExperiment/templates/VisualEditorTourExperiment_content.mustache',
+				}
+			}).done(this._setupTour.bind(this));
+		}
+
+		Tour.prototype.destroyStep = function(step) {
+			var tourStepData = this.steps[step],
+				$element = tourStepData ? tourStepData.$element : null;
+
+			if ($element) {
+				$element.popover('destroy');
+			}
+		}
+
+		Tour.prototype.openStep = function(step) {
+			var tourStepData = this.steps[step],
+				$element = tourStepData ? tourStepData.$element : null;
+
+			if (!$element) {
+				return;
+			}
+
+			$element.popover({
+				content: tourStepData.content,
+				html: true,
+				placement: this.tourConfig[step].placement,
+				trigger: 'manual'
+			});
+
+			$element.popover('show');
+
+			track({
+				action: tracker.ACTIONS.IMPRESSION,
+				label: this.labelPrefix + 'tour-step-' + this.step
+			});
+		}
+
+		Tour.prototype.nextHandle = function() {
+			this._setDisabled();
+			this.next();
+			track({
+				action: tracker.ACTIONS.CLICK,
+				label: this.labelPrefix + 'next-go-to-' + this.step
+			});
+		}
+
+		Tour.prototype.next = function() {
+			if (this.step === this.steps.length - 1) {
+				this.destroyStep(this.step);
+				track({
+					action: tracker.ACTIONS.CLICK,
+					label: this.labelPrefix + 'tour-complete'
+				});
+				return;
+			}
+			this.destroyStep(this.step);
+			this.openStep(++this.step);
+		}
+
+		Tour.prototype.prevHandle = function() {
+			this.destroyStep(this.step);
+			this.openStep(--this.step);
+			track({
+				action: tracker.ACTIONS.CLICK,
+				label: this.labelPrefix + 'next-go-to-' + this.step
+			});
+		}
+
+		Tour.prototype.close = function() {
+			this._setDisabled();
+			this.destroyStep(this.step);
+			track({
+				action: tracker.ACTIONS.CLICK,
+				label: this.labelPrefix + 'close'
+			});
+			track({
+				action: tracker.ACTIONS.CLICK,
+				label: this.labelPrefix + 'close-' + this.step
```

Comment:
Do we need two tracking events for this or can we just sum the `close-<step>` events?